packages/components/src/internal/components/search/utils.ts

```diff
@@ -343,3 +346,112 @@ export function getUpdateFilterExpressionFilter(
 
     return filter;
 }
+
+// this util is only for string field type
+export function getCheckedFilterValues(filter: Filter.IFilter, allValues: string[]): string[] {
+    if (!filter || !allValues)
+        // if no existing filter, check all values by default
+        return allValues;
+
+    const filterUrlSuffix = filter.getFilterType().getURLSuffix();
+    const filterValues = getFilterValuesAsArray(filter);
+
+    switch (filterUrlSuffix) {
+        case '':
+        case 'any':
+            return allValues;
+        case 'isblank':
+            return [EMPTY_VALUE_DISPLAY];
+        case 'isnonblank':
+            return allValues.filter(value => value !== EMPTY_VALUE_DISPLAY && value !== ALL_VALUE_DISPLAY);
+        case 'neq':
+        case 'neqornull':
+            return allValues.filter(value => value !== filterValues[0] && value !== ALL_VALUE_DISPLAY);
+        case 'eq':
+        case 'in':
+            return filterValues;
+        case 'notin':
+            return allValues.filter(value => filterValues.indexOf(value) === -1 && value !== ALL_VALUE_DISPLAY);
+        default:
+            return [];
+    }
+}
+
+export function getUpdatedCheckedValues(
+    allValues: string[],
+    newValue: string,
+    check: boolean,
+    oldFilter: Filter.IFilter,
+    uncheckOthers?: boolean
+): string[] {
+    if (uncheckOthers) return [newValue];
+
+    const oldCheckedValues = getCheckedFilterValues(oldFilter, allValues);
+    let newCheckedValues = [...oldCheckedValues];
```

Comment:
I believe this can be `const` instead of `let`

Reply:
it does get assigned a new array, not just operated it, so needs to be let